src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Util.java

```diff
@@ -187,4 +190,23 @@ static String toString(byte[] b) {
         return sb.toString();
     }
 
+    // returns true if successfully cancelled
+    static boolean trySessionCancel(Token token, Session session, long flags)
+            throws ProviderException {
+        boolean status = false;
+        if (token.p11.getVersion().major == 3) {
+            try {
+                token.p11.C_SessionCancel(session.id(), flags);
+                status = true;
```

Comment:
Or you can just `return true` here and `return false` at the end.

Reply:
Sure, I like this better.